src/psyclone/tests/domain/nemo/transformations/nemo_allarrayrange2loop_trans_test.py

```diff
@@ -121,6 +121,55 @@ def test_apply_multi_assignments():
     assert expected in result
 
 
+def test_apply_with_structures(fortran_reader, fortran_writer):
+    '''Check that the PSyIR is transformed as expected when the
+    expressions contain a mix of arrays and structures in the same
+    reference.
+
+    '''
+    trans = NemoAllArrayRange2LoopTrans()
+
+    # The inner dimension is already set
+    psyir = fortran_reader.psyir_from_source('''
+    subroutine test
+        use my_variables
+        integer, parameter :: constant = 3
+        integer :: jk
+        base%field(constant)%array(:,:,jk) = 1
+    end subroutine test
+    ''')
+    assignment = psyir.walk(Assignment)[0]
+    trans.apply(assignment)
+    result = fortran_writer(assignment)
+    assert "base%field(constant)%array(ji,jj,jk) = 1" in result
+
+    # The outer dimension is already set
+    psyir = fortran_reader.psyir_from_source('''
+    subroutine test
+        use my_variables
+        integer, parameter :: jf = 3, jpi = 3, jpim1 = 1
+        ptab(jf)%pt2d(jpi,:,:) = ptab(jf)%pt2d(jpim1,:,:)
```

Comment:
I'm a bit confused about 'outer' and 'inner' here - I would have thought this was 'inner' as the dimension that is set is the one that is contiguous in memory?

Reply:
You are right, I had a lapsus with C